mlrun/model_monitoring/db/tsdb/tdengine/tdengine_connector.py

```diff
@@ -260,13 +325,69 @@ def read_predictions(
         start: datetime,
         end: datetime,
         aggregation_window: typing.Optional[str] = None,
+        limit: typing.Optional[int] = None,
+        agg_funcs: typing.Optional[list] = None,
     ) -> typing.Union[
         mm_schemas.ModelEndpointMonitoringMetricValues,
         mm_schemas.ModelEndpointMonitoringMetricNoData,
     ]:
-        raise NotImplementedError
+        if (agg_funcs and not aggregation_window) or (
+            aggregation_window and not agg_funcs
+        ):
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Both `aggregation_window` and `agg_funcs` must be provided or neither of them."
```

Comment:
I think the convention is no capitalization and no period at the end of the sentence
```suggestion
                "both or neither of `aggregation_window` and `agg_funcs` must be provided"
```